lib/scan_result.dart

```diff
@@ -14,49 +14,66 @@ abstract class _ScanResultMetadata {
 
 /// A scan result emitted by the scanning operation, containing [Peripheral] and [AdvertisementData].
 class ScanResult {
-  Peripheral peripheral;
+  final Peripheral peripheral;
 
   /// Signal strength of the peripheral in dBm.
-  int rssi;
+  final int rssi;
 
   /// An indicator whether the peripheral is connectable (iOS only).
-  bool isConnectable;
+  late final bool isConnectable;
```

Comment:
now, that's why it's great that you review this, ofc you are much more aware of the edge cases and behaviours of you own library.
What do you think we move it to an enum instead since it's 3.0 milestone anyways. 
Like `final Connectivity connectivity;` with 3 states:
```Connectivity.noInfo, Connectivity.notConnectable, Connectivity.Connectable```
I'm actually okay with the nullable version as well, but I'm also anxious, people would merge the cases of false and null creating bad logic accidently. It's a common mistake. Thus I generally wouldn't prefer a nullable `bool?` field anywhere.

As the semantic difference of null and false is often unclear and some libs also use them interchangeably, meaning they sloppily emit null instead of false again JS's fault.. the truthy falsy fiasco...

Reply:
Enum sounds OK. I've slapped the milestone 3.0 on this, but I'm actually inclined to release this as 2.4.0, as I don't see any breaking changes, so I'll create a separate issue for the enum.

Reply:
https://github.com/Polidea/FlutterBleLib/issues/588